pkg/stanza/fileconsumer/internal/reader/factory.go

```diff
@@ -74,7 +74,6 @@ func (f *Factory) NewReaderFromMetadata(file *os.File, m *Metadata) (r *Reader,
 		initialBufferSize:    f.InitialBufferSize,
 		maxLogSize:           f.MaxLogSize,
 		decoder:              decode.New(f.Encoding),
-		lineSplitFunc:        f.SplitFunc,
```

Comment:
I have removed this line, as `lineSplitFunc` is being reassigned below in line 106.